### PR TITLE
Wishlist

### PR DESCRIPTION
**Pull request by Jia-A**
This pull request contains wishlist page, and required contexts for it.
Please review it.

### CODE REVIEW

File: src/dot-components/Wishlist/wishlist.jsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                <img src= {item.image} alt={item.title} class="card-image"/>
+                <div class="main-info-cont">
+                    <h3 class="info-head">{item.title}</h3>
+                    <p class="sm-txt">{item.author}</p>
+                    <p className="sub-info">{item.ratings}<span><i class="fas fa-star"></i></span></p>
+                    <p class="sub-info">Rs. {item.price}</p>
+                </div>
+                {/* <div class="sub-info-cont">
+                    <p class="sub-info">Rs. {item.price}</p>
+                </div> */}
```
Comment (shuklahimanshu81) [inline src/dot-components/Wishlist/wishlist.jsx:28]: remove the comments

File: src/dot-components/Wishlist/wishlist.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                    {cartState.cart.find((product) => product._id===item._id) ? 
+                    <button class="btn primary-btn" onClick={ ()=> cartDispatch({type : "Increase_quantity", payload : item})}>Increase quantity</button> : 
+                    <button class="btn primary-btn" onClick={ ()=> cartDispatch({type : "Add_to_cart", payload : item})}>Move to cart</button> }
+                    <button class="button read-btn" onClick = { ()=> wishDispatch({type : "Remove_from_wishlist", payload : item})}>Remove <i class="fas fa-heart wish-icon"></i></button>
+                </div>
+                </article>
+            )}
+            
+            
+        </div>
```
Comment (shuklahimanshu81) [inline src/dot-components/Wishlist/wishlist.jsx]: Remove extra spaces.

File: src/dot-components/Wishlist/wishlist.css [hunk 3]
```diff
@@ -0,0 +1,56 @@
+@import url('https://fonts.googleapis.com/css2?family=Montserrat:ital,wght@0,300;0,400;0,500;0,900;1,300&display=swap');
+
+body{
+    font-family: "Montserrat", sans-serif;
+}
+.wishlist-container{
+    gap: 40px;
```
Comment (shuklahimanshu81) [inline src/dot-components/Wishlist/wishlist.css]: better to use relative units

File: src/dot-components/Wishlist/wishlist.css [hunk 4]
```diff
@@ -0,0 +1,56 @@
+@import url('https://fonts.googleapis.com/css2?family=Montserrat:ital,wght@0,300;0,400;0,500;0,900;1,300&display=swap');
+
+body{
+    font-family: "Montserrat", sans-serif;
+}
```
Comment (shuklahimanshu81) [inline src/dot-components/Wishlist/wishlist.css]: importing fonts in root is better will save the time and also you will not have to repeat.

File: src/Context/cartContext.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                        };
+                    case 'Decrease_quantity':
+                        return {...cartState,
+                        cart: 
+                            cartState.cart.map((item) => item.id === action.payload.id ? 
+                                {...item, quantity: item.quantity > 1 ? item.quantity - 1: item.quantity} : item)
+                        };
+                        case 'Remove_from_cart':
+                            return {
+                                ...cartState,
```
Comment (shuklahimanshu81) [inline src/Context/cartContext.js]: indentation is bit weird .

**Review by shuklahimanshu81 — APPROVED** [review]
Everything else is good . Good to merge

File: src/dot-components/Wishlist/wishlist.css (hunk 3 above)
Comment (Jia-A) [reply src/dot-components/Wishlist/wishlist.css]: Oh yeah, forgot to convert it. Will do that.

File: src/dot-components/Wishlist/wishlist.css (hunk 4 above)
Comment (Jia-A) [reply src/dot-components/Wishlist/wishlist.css]: That will work across the whole project? Okay will change this.

File: src/Context/cartContext.js (hunk 5 above)
Comment (Jia-A) [reply src/Context/cartContext.js]: Okay I will format it.

File: src/Context/cartContext.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import { createContext, useContext, useReducer } from "react";
+
+const CartContext = createContext();
+
+const cartFunction = (cartState, action) =>{
+    switch(action.type) {
+        case 'Add_to_cart':
```
Comment (abhij1607) [inline src/Context/cartContext.js]: make these case values in all capitals

**Review by abhij1607 — COMMENTED** [review]
Right logic used

Suggestions-
wishlist and cart context can be merged into single context
use formatted code
convert html to jsx such as class to className

**Pull request merged**